### DISCUSSION TITLE
4 port emulation on ODI DFP-34X-2C2

### DISCUSSION THREAD

**Discussion by rajil** [Anime4000/RTL960x#185] [General]
I am planning to use ODI DFP-34X-2C2 with pfsense and was curious about 4 port emulation. My current ONU is Huawei HG8145V5 which has 4 physical ports and it is possible to bridge each of the ports separately.

I am wondering wow does the 4 port emulation work on SFP. I suppose a single NIC is exposed to the pfsense router which carries multiple VLANS. Does 4 port emulation allow you to specify what VLANs need to be exposed?

**Comment by Anime4000** [comment]
4 port emulation or known as `EntityId` Mapping.

OLT can assign which interface is Internet or IPTV or etc...
if ONU has WiFi, OLT also can assign to WiFi, like using GPON as WiFi AP only

by default, ODI/VSOL lazy map all `EntityId` to single interface.
this useful for ISP assigning each LAN different service and different VLAN
issue when different LAN is different service but no VLAN, thus ODI will merge cause service conflict...

maybe you need use ODI IGD firmware instead, with known VLAN, you can assign different VLAN into single SFP LAN interface
